### PR TITLE
Adding spec

### PR DESCRIPTION
**Pull request by youyifong**

### CODE REVIEW

File: data_clean/verification/specifications.rmd [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  \item Define race, a factor variable that has the following levels:   "White", "Black or African American",
+  "Asian", "American Indian or Alaska Native",
+  "Native Hawaiian or Other Pacific Islander", "Multiracial",
+  "Other", "Not reported and unknown"
+
+  \item Define WhiteNonHispanic, a binary variable that takes value 1 if race is White and ethnicity is not Hispanic, 0 if race is not "white or unknown" OR ethnicity is Hispanic, and NA otherwise
+  
+  \item Define URM, a binary variable that takes value 1 if WhiteNonHispanic is 0, and 0 otherwise including when WhiteNonHispanic is NA
+  
+  \item Define Bstratum, an integer variable to index randomization strata. For Moderna, there are three strata: age >= 65, age < 65 and HighRiskInd==1, age < 65 and HighRiskInd==0
```
Comment (thebioengineer) [inline data_clean/verification/specifications.rmd:38]: Please indicate which strata are which integer

File: data_clean/verification/specifications.rmd [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  
+  \item Define demo.stratum, an integer variable to index the demographics strata. For Moderna, there are six strata: URM [1, 0] x Bstratum
```
Comment (thebioengineer) [inline data_clean/verification/specifications.rmd:40]: Please indicate which combination of URM/Bstratum are which integer

File: data_clean/verification/specifications.rmd [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  
+  \item Define tps.stratum, an integer variable to index correlates sampling strata. For Moderna, there are 24 strata: [placebo/baseline negative, placebo/baseline positive, vaccine/baseline negative, vaccine/baseline positive] x demo.stratum
+
+  \item Define Wstratum, an integer variable to index strata for computing weights. For Moderna, there are 28 strata: if not case (EventIndPrimaryD29==1): demo.stratum, else: [25 - placebo/baseline negative, 26 - placebo/baseline positive, 27 - vaccine/baseline negative, 28 - vaccine/baseline positive]
```
Comment (thebioengineer) [inline data_clean/verification/specifications.rmd:44]: should demo.stratum be tps.stratum?

File: data_clean/verification/specifications.rmd [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  
+  \item Define wt.2, a double variable for D29 correlates analyses. The weights equal the sampling probabilities of TwophasesampInd.2==1 within each Wstratum. This variable is only defined for subjects satisfying Perprotocol == 1 and EventTimePrimaryD57>=7 | EventTimePrimaryD29 >= 7 and EventTimePrimaryD29 <= 13 and Fullvaccine == 1 and set to NA otherwise
+  
+  \item Define wt.subcohort, a double variable for immunogenicity analyses that use subcohort only and are not enriched by cases outside subcohort. The weights equal the sampling probabilities of TwophasesampInd==1 and SubcohortInd==1 within each tps.stratum. This variable is only defined for subjects satisfying Perprotocol == 1 and EventTimePrimaryD57>=7 and set to NA otherwise
+  
+  \item Carry out single imputation using the R package mice for the subjects satisfying TwophasesampInd==1. Since baseline, D29 and D57 binding to Spike and RBD are required in this subset of people, the imputation is intended to fill in missing marker values for neutralization assays at these three time points. (\_common.R defines a vector variable, markers, for this purpose.)
```
Comment (thebioengineer) [inline data_clean/verification/specifications.rmd:52]: @youyifong - in order to be considered for TwophasesampleInd, participants must not have any missing binding to spike or RBD. could you clarify what you mean here?

**Comment by youyifong** [comment]
I meant they need to have
        BbindSpike, Day29bindSpike, Day57bindSpike,
        BbindRBD, Day29bindRBD, Day57bindRBD

File: data_clean/verification/specifications.rmd [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  
+  \item Define wt.subcohort, a double variable for immunogenicity analyses that use subcohort only and are not enriched by cases outside subcohort. The weights equal the sampling probabilities of TwophasesampInd==1 and SubcohortInd==1 within each tps.stratum. This variable is only defined for subjects satisfying Perprotocol == 1 and EventTimePrimaryD57>=7 and set to NA otherwise
+  
+  \item Carry out single imputation using the R package mice for the subjects satisfying TwophasesampInd==1. Since baseline, D29 and D57 binding to Spike and RBD are required in this subset of people, the imputation is intended to fill in missing marker values for neutralization assays at these three time points. (\_common.R defines a vector variable, markers, for this purpose.) 
+  
+  \item Define Delta57overBxxx, Delta29overBxxx, and Delta57over29xxx for each xxx in the variable assays. They are the log10 fold rise markers.
```
Comment (thebioengineer) [inline data_clean/verification/specifications.rmd:54]: Could you define how to approach negative fold values?

**Pull request merged**